src/reporter.test.ts

```diff
@@ -589,6 +588,97 @@ describe('reporter', () => {
     expect(reporter.forgeAndScrubRequestBody([], 100)).toEqual(undefined);
   });
 
+  test('scrubSpans missing http fields', () => {
+    const spans = [
+      {
+        info: {
+          httpInfo: {
+            host: 'host',
+            request: {}, // missing body and headers
+            response: {}, // missing body and headers
+          },
+        },
+      },
+      {
+        info: {
+          // missing request and response and host
+          httpInfo: {},
+        },
+      },
+      { info: {} }, // missing httpInfo
+      {}, // missing info
+      {
+        info: {
+          httpInfo: {
+            host: 'host',
+            // missing headers
+            request: {
+              body: '',
+            },
+            // missing body
+            response: {
+              headers: {},
+            },
+          },
+        },
+      },
+      {
+        info: {
+          httpInfo: {
+            // missing host
+            request: {
+              body: '',
+              headers: {},
+            },
+            response: {
+              body: '',
+              headers: {},
+            },
+          },
+        },
+      },
+    ];
+    scrubSpans(spans);
+    expect(spans.length).toEqual(spans.length);
+  });
+
+  test('scrubSpans missing http fields (stepFunction)', () => {
+    const spans = [
+      {
+        info: {
+          traceId: {
+            Root: 'Root',
+            Parent: 'Parent',
+            Sampled: '0',
+            transactionId: 'transactionId',
+          },
+          httpInfo: { host: 'StepFunction' },
+          resourceName: 'StepFunction',
+          messageId: 'messageId',
+        },
+      },
+    ];
+    scrubSpans(spans);
+    expect(spans.length).toEqual(spans.length);
+  });
+
+  test('scrubSpans should not fail after throwing an error', () => {
+    jest
+      .spyOn(utils, 'shouldScrubDomain')
+      .mockReturnValueOnce(true)
+      .mockReturnValueOnce(true)
+      .mockImplementationOnce(() => {
+        throw new Error('Error');
+      });
+    const spans = [
+      { info: { httpInfo: { request: {}, response: {} } } },
+      { info: { httpInfo: { request: {}, response: {} } } },
+      { info: { httpInfo: { request: {}, response: {} } } },
+    ];
+    scrubSpans(spans);
```

Comment:
If the scrub failed in the third span, what do you send there?
can you check the return value?